src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,51 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private BulldozerProducerImpl bulldozerProducer;
+    private ExcavatorProducerImpl excavatorProducer;
+    private TruckProducerImpl truckProducer;
+
+    public MachineServiceImpl() {
+    }
+
+    public MachineServiceImpl(BulldozerProducerImpl bulldozerProducer,
+                              ExcavatorProducerImpl excavatorProducer,
+                              TruckProducerImpl truckProducer) {
+        this.bulldozerProducer = bulldozerProducer;
+        this.excavatorProducer = excavatorProducer;
+        this.truckProducer = truckProducer;
+    }
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machineList = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            machineList.addAll(bulldozerProducer.get());
+        } else if (type.equals(Excavator.class)) {
+            machineList.addAll(excavatorProducer.get());
+        } else {
+            machineList.addAll(truckProducer.get());
+        }
+        return machineList;
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. This will allow you to directly return the result of the producer's `get()` method without the need of an intermediate list. For example, `List<? extends Machine> machines = bulldozerProducer.get(); return new ArrayList<>(machines);`